### PR TITLE
Identifier for Retry Pages

### PR DESCRIPTION
**Pull request by shashimalcse**
### Proposed changes in this pull request

add sessionDataKey, an identifier for find a correlation between the original authentication flow from the last redirection to the retry page. 

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractLocalApplicationAuthenticator.java [hunk 1]
```diff
@@ -85,8 +86,9 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
                 } catch (AuthenticationFailedException e) {
                     if (isAccountLocked(context)) {
                         try {
-                            String redirectUrl = getRedirectUrlOnAccountLock(context, response);
-                            response.sendRedirect(redirectUrl);
+                            String status = FrameworkConstants.ACCOUNT_LOCKED_MSG;
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/AbstractLocalApplicationAuthenticator.java:89]: If getRedirectUrlOnAccountLock method is not used anywhere else, let's remove it

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java [hunk 2]
```diff
@@ -109,6 +109,8 @@ public abstract class FrameworkConstants {
 
     public static final String REQUEST_PARAM_SP = "sp";
     public static final String REQUEST_PARAM_ERROR_KEY = "errorKey";
+    public static final String REQUEST_PARAM_SESSION_DATA_KEY = "sessionDataKey";
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkConstants.java:112]: This should be defined already, please check

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java [hunk 3]
```diff
@@ -486,7 +486,8 @@ private void handleAccountLockLoginFailure(String retryPage, AuthenticationConte
             boolean showAuthFailureReason = true;
             retryPage = FrameworkUtils.appendQueryParamsStringToUrl(retryPage,
                     "sp=" + context.getServiceProviderName());
-                                                                    
+            retryPage = FrameworkUtils.appendQueryParamsStringToUrl(retryPage,
+                    "authFlowId=" + context.getContextIdentifier());
```
Comment (senthalan) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java:490]: let's use this constant REQUEST_PARAM_AUTH_FLOW_ID

**Review by senthalan — APPROVED** [review]

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java [hunk 4]
```diff
@@ -486,7 +486,9 @@ private void handleAccountLockLoginFailure(String retryPage, AuthenticationConte
             boolean showAuthFailureReason = true;
             retryPage = FrameworkUtils.appendQueryParamsStringToUrl(retryPage,
                     "sp=" + context.getServiceProviderName());
-                                                                    
+            retryPage = FrameworkUtils.appendQueryParamsStringToUrl(retryPage,
+                    String.format("%s=", FrameworkConstants.REQUEST_PARAM_AUTH_FLOW_ID)
```
Comment (mefarazath) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/JITProvisioningPostAuthenticationHandler.java:490]: Any reason for using formatting and concatenation both here?
If we are using String.format why not use "%s=%s"

**Pull request merged**